app/models/survey.rb

```diff
@@ -0,0 +1,186 @@
+class Survey < Assignment
+  enum question_kind: [:yes_no, :true_false, :multiple_choice, :numeric, :text]
+
+  before_create :set_survey_grader
+
+  def sub_not_following_related?(user)
```

Comment:
This doesn't go here, it should go in Pset.  The prohibited workflow is (1) submit a problem set, (2) do a self-eval about that problem set, (3) now that you've seen the questions, go back and resubmit the problem set with fixes.  So this method gets called on the problem set, to see whether anything has been submitted to (survey) assignments that think they're related to this one.